src/utils/solana.js

```diff
@@ -6,10 +6,20 @@ const connection = new Connection(clusterApiUrl("mainnet-beta"));
 export const ANTI_TOKEN_MINT = new PublicKey(process.env.NEXT_PUBLIC_ANTI_TOKEN_MINT);
 export const PRO_TOKEN_MINT = new PublicKey(process.env.NEXT_PUBLIC_PRO_TOKEN_MINT);
 
-export const checkTokenBalance = async (walletPublicKey, mint) => {
+export const getTokenBalance = async (walletPublicKey, mint) => {
   const accounts = await connection.getParsedTokenAccountsByOwner(
     walletPublicKey,
     { mint }
   );
-  return accounts.value.length > 0 && accounts.value[0].account.data.parsed.info.tokenAmount.uiAmount > 0;
+  return accounts.value?.reduce((memo, item) => {
+    return memo + item?.account?.data.parsed.info.tokenAmount.uiAmount || 0
```

Comment:
would suggest to wrap the check for uiAmount

```suggestion
    return memo + (item?.account?.data.parsed.info.tokenAmount.uiAmount || 0);
```